File: src/main/java/me/kuku/yuq/dao/MotionDao.java
package me.kuku.yuq.dao;

import com.icecreamqaq.yudb.YuDao;
import com.icecreamqaq.yudb.jpa.annotation.Dao;
import com.icecreamqaq.yudb.jpa.annotation.Execute;
import com.icecreamqaq.yudb.jpa.annotation.Select;
import me.kuku.yuq.entity.StepEntity;

import java.util.List;

@Dao
public interface MotionDao extends YuDao<StepEntity, Integer> {
    StepEntity findByQQ(Long qq);
    @Select("from MotionEntity")
    List<StepEntity> findAll();
    @Execute("delete from MotionEntity where qq = ?0")
    void delByQQ(Long qq);
}

File: src/main/kotlin/me/kuku/yuq/dao/MotionDao.kt
package me.kuku.yuq.dao

import com.icecreamqaq.yudb.YuDao
import com.icecreamqaq.yudb.jpa.annotation.Dao
import com.icecreamqaq.yudb.jpa.annotation.Execute
import com.icecreamqaq.yudb.jpa.annotation.Select
import me.kuku.yuq.entity.MotionEntity

@Dao
interface MotionDao: YuDao<MotionEntity, Int>{
    fun findByQQ(qq: Long): MotionEntity?
    @Select("from MotionEntity")
    fun findAll(): List<MotionEntity>
    @Execute("delete from MotionEntity where qq = ?1")
    fun delByQQ(qq: Long)
}